### ISSUE TITLE
In a triple-quoted string containing contiguous blank lines, an interpolated value causes loss of blank lines

### ISSUE THREAD

**Issue by rudifa** [nicklockwood/SwiftFormat#1078]
That is, multiple blank lines following a `\(someExpression)` get squashed into a single blank line.

Example: when you  `swiftformat` a file containing this construct ...

```
    var interpolated = "XXX"
    var shortSample2: String {
        return """
        AAA
        \(interpolated)




        BBB



        """
    }
```

... the 4 blank lines following `\(interpolated)` get squashed  into a single blank line:

```
    var shortSample2: String {
        return """
        AAA
        \(interpolated)

        BBB



        """
    }
```

This is so with the current release `0.48.18`, as well as with the `0.49.0` from the `develop` branch.

The attached .zip file contains `SampleMultilineStrings.swift`  and  `run-test.sh` to aid in the investigation.

The `SampleMultilineStrings.swift` contains several sample strings similar to `shortSample2` above.
When run as a swift script it prints the line counts for the sample strings:

```
investigate-squashed-blank-lines % cp SampleMultilineStrings.swift formatted-SampleMultilineStrings.swift                                                          

investigate-squashed-blank-lines % swiftformat formatted-SampleMultilineStrings.swift
Running SwiftFormat...
warning: No Swift version was specified, so some formatting features were disabled. Specify the version of Swift you are using with the --swiftversion option, or by adding a .swift-version file to your project.
SwiftFormat completed in 0.1s.
1/1 files formatted.

investigate-squashed-blank-lines % ./formatted-SampleMultilineStrings.swift

running formatted-SampleMultilineStrings.swift...
realLifeSample1 OK
realLifeSample2 FAILED: expected 32 lines, actual 26
shortSample1 OK
shortSample2 FAILED: expected 10 lines, actual 7
shortSample3 FAILED: expected 10 lines, actual 9
shortSample4 OK
shortSample5 FAILED: expected 10 lines, actual 9
shortSample6 OK

``` 

The script `run-test.sh` mechanizes above sequence, for repetitive tests.

[investigate-squashed-blank-lines.zip](https://github.com/nicklockwood/SwiftFormat/files/7563473/investigate-squashed-blank-lines.zip)

**Comment by rudifa** [comment]
@nicklockwood I am a little confused, can you explain what I see?

On my local branch develop-rf, off the branch `* 486a7262 (upstream/develop, develop)` ... 

1. I added a test function in the file `Tests/RulesTests+Linebreaks.swift`

After a future fix for the loss-of-blank-lines bug that I reported, I would expect this test to pass.

```
    func testConsecutiveBlankLinesAfterInterpolation() {
        let interpolated = "XXX"
        let input = """
        AAA
        \(interpolated)



        BBB



        """
        let output = """
        AAA
        XXX



        BBB



        """
        testFormatting(for: input, output, rule: FormatRules.consecutiveBlankLines)
    }

    func testLintingConsecutiveBlankLinesReportsCorrectLine() { 
``` 

2. As I launch `LinebreakTests` in Xcode, on the scheme `SwiftFormat (Framework)`, `Tests/RulesTests+Linebreaks.swift` itself gets reformatted, in a way consistent to what I reported: three blank lines following `\(interpolated)` are squashed into one.

* I find this reformat undesired, I would prefer the test file to remain unmodified.

```
    func testConsecutiveBlankLinesAfterInterpolation() {
        let interpolated = "XXX"
        let input = """
        AAA
        \(interpolated)

        BBB



        """
        let output = """
        AAA
        XXX



        BBB



        """
        testFormatting(for: input, output, rule: FormatRules.consecutiveBlankLines)
    }

    func testLintingConsecutiveBlankLinesReportsCorrectLine() {
```

3. Test fails, as expected, and displays two error messages.

* However, in the second message the input string is not as seen in (2):
- input lacks the 3 blank lines that follow `BBB` in the file.
- output has only 2 blank lines that follow `BBB`, instead of 3.

```
XCTAssertEqual failed: ("[:4:1: warning: (consecutiveBlankLines) Replace consecutive blank lines with a single blank line., :7:1: warning: (consecutiveBlankLines) Replace consecutive blank lines with a single blank line., :8:1: warning: (consecutiveBlankLines) Replace consecutive blank lines with a single blank line.]") is not equal to ("[]")


XCTAssertEqual failed: ("AAA
XXX

BBB
") is not equal to ("AAA
XXX



BBB


")
```

**Comment by rudifa** [comment]
I submitted a pull request containing a fix for this problem, `Develop rudifa #1079`

**Comment by nicklockwood** [comment]
@rudifa landed in 0.49.0
